crates/oxc_linter/src/rules/eslint/no_restricted_imports.rs

```diff
@@ -1832,6 +2600,270 @@ fn test() {
         // ),
     ];
 
+    let fail_typescript = vec![
+        ("import foo from 'import1';", Some(serde_json::json!(["import1", "import2"]))),
+        // ("import foo = require('import1');", Some(serde_json::json!(["import1", "import2"]))),
```

Comment:
@camc314 should we support this?
Notice the `import` at the start. There is no entry in `ModuleRecord` for it.
I searched a bit on the internet and it looks like this is a old syntax. 

There is 2 options how to solve this:
- fix `ModuleRecord` to implement this use case 
- loops to the AST nodes and find the require

I am not that good in typescript to understand what the best solution for the absurd syntax

Reply:
the team worked on some semantic about this. So it is still needed. Need to rethink the implementation